RHDEVDOCS-6182: Creating content for the custom TLS certificates (#85044)
openshift/openshift-docs: modules/gitops-argo-cd-installation.adoc

====

.. Optional: Modify the route TLS termination policy by configuring the `.spec.server.route.tls` field of the Argo CD CR.
+
[NOTE]
====

> When configuring custom TLS certificates for Argo CD Server route, avoid using the `.spec.server.route.tls.key` & `.spec.server.route.tls.certificate` fields. Use the `.spec.server.route.tls.externalCertificate` instead. For more information about configuring a route for custom TLS certificate, see examples in link:https://argocd-operator.readthedocs.io/en/latest/usage/routes/#custom-tls-certificates[Custom TLS certificates for Routes].

Reviewer: ```suggestion
When configuring custom TLS certificates for Argo CD Server route, avoid using the `.spec.server.route.tls.key` and `.spec.server.route.tls.certificate` fields. Use the `.spec.server.route.tls.externalCertificate` field instead. For more information about configuring a route for custom TLS certificate, see examples in link:https://argocd-operator.readthedocs.io/en/latest/usage/routes/#custom-tls-certificates[Custom TLS certificates for Routes].
```

> When configuring custom TLS certificates for Argo CD Server route, you must not use the `.spec.server.route.tls.key` & `.spec.server.route.tls.certificate` fields. Use the `.spec.server.route.tls.externalCertificate` instead. For more information about configuring a route for custom TLS certificate, see examples in link:https://argocd-operator.readthedocs.io/en/latest/usage/routes/#custom-tls-certificates[Custom TLS certificates for Routes].

Reviewer 2: ```suggestion
When configuring custom TLS certificates for Argo CD Server route, it is not recommended to use the `.spec.server.route.tls.key` & `.spec.server.route.tls.certificate` fields. Use the `.spec.server.route.tls.externalCertificate` instead. For more information about configuring a custom TLS certificate for ArgoCD routes, see examples in link:https://argocd-operator.readthedocs.io/en/latest/usage/routes/#custom-tls-certificates[Custom TLS certificates for Routes].
```

It's not a strict requirement but a suggestion so I think we should use `not recommended` or something similar instead of `must not`

Author: Fixed as per our discussion in Slack.
